### PR TITLE
Extract mapping logic into its own interface

### PR DESCRIPTION
**Pull request by jhorstmann**
This allows optimized implementation that only extract a single property
from an event, or for example to map directly to clojure datastructures.

### CODE REVIEW

File: fahrschein/src/main/java/org/zalando/fahrschein/MappingEventReader.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    events.add(eventClass.cast(eventIterator.next()));
+                } else {
+                    break;
+                }
+            } catch (RuntimeException e) {
+                final Throwable cause = e.getCause();
+                if (cause instanceof JsonMappingException) {
+                    onMappingException((JsonMappingException) cause);
+                } else if (cause instanceof IOException) {
+                    throw (IOException)cause;
```
Comment (lukasniemeier-zalando) [inline fahrschein/src/main/java/org/zalando/fahrschein/MappingEventReader.java]: whitespace :wink:

File: fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java [hunk 2]
```diff
@@ -4,39 +4,31 @@
...
-import com.fasterxml.jackson.databind.ObjectReader;
 import com.fasterxml.jackson.databind.ObjectWriter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.zalando.fahrschein.domain.Batch;
-import org.zalando.fahrschein.domain.Cursor;
-import org.zalando.fahrschein.domain.Lock;
-import org.zalando.fahrschein.domain.Partition;
-import org.zalando.fahrschein.domain.Subscription;
+import org.zalando.fahrschein.domain.*;
```
Comment (lukasniemeier-zalando) [inline fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java]: on purpose?

File: fahrschein/src/main/java/org/zalando/fahrschein/JsonParserHelper.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import javax.annotation.Nullable;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+
+class JsonParserHelper {
+    private JsonParserHelper() {
+    }
+
+    static void expectToken(JsonParser jsonParser, JsonToken expectedToken) throws IOException {
```
Comment (lukasniemeier-zalando) [inline fahrschein/src/main/java/org/zalando/fahrschein/JsonParserHelper.java]: does it make sense to mark this as `public`?

Comment (jhorstmann) [reply fahrschein/src/main/java/org/zalando/fahrschein/JsonParserHelper.java]: Should be used only internally. Also could not think of a better name than `JsonParserHelper` but for a non-public class this is probably not that important.

**Comment by jhorstmann** [comment]
:+1:

**Pull request merged**